src/components/bcros/businessDetails/Links.vue

```diff
@@ -126,8 +135,26 @@ const downloadBusinessSummary = async (): Promise<void> => {
 
 /** Creates a draft filing and navigates to the Create UI to file a company dissolution filing. */
 const dissolveBusiness = async (): Promise<void> => {
-  // To be implemented in ticket #23467
-  await new Promise<void>((resolve) => { resolve() })
+  const payload = {
+    custodialOffice: currentBusinessAddresses.value?.registeredOffice,
+    dissolutionType: 'voluntary'
+  }
+  const response = await filings.createFiling(
+    currentBusiness.value,
+    FilingTypes.DISSOLUTION,
+    payload,
+    true
+  )
+
+  await new Promise<void>((resolve, reject) => {
+    if (response.error?.value) {
+      console.error('Filing error', response.error.value)
+      reject(new Error('Failed to create filing'))
+    } else {
+      goToCreatePage('/dissolution-define-dissolution', { id: currentBusiness.value.identifier })
+      resolve()
+    }
+  })
```

Comment:
In the `dissolveCompany()` in the old dashboard https://github.com/bcgov/business-filings-ui/blob/b5e22fbb8945288b42fd1f0e23d62a0479a8ac08/src/App.vue#L835-L846, before navigating to a new page, it checks for `if (!draftDissolution || isNaN(draftDissolutionId))`, I am not sure if our `response.error?.value` can cover all the failure cases.

Reply:
It should cover all cases but I've added the check for completeness sake